### PR TITLE
Implement source event `DownloadFinishedEvent`

### PR DESCRIPTION
**Pull request by 123mpozzi**
## Description

Implement source event DownloadFinishedEvent to indicate once a download was finished.

## Changes

- Add `DownloadFinishedEvent`
- Add `HttpRequestType`

## Testing
Apply the following patch:
[testing.patch](https://github.com/bitmovin/bitmovin-player-react-native/files/13166477/testing.patch)

Check for:
- Event being emitted correctly

## Checklist
- [x] 🗒 `CHANGELOG` entry

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 - `LiveConfig.minTimeshiftBufferDepth` to control the minimum buffer depth of a stream needed to enable time shifting.
 - `Player.buffer` to control buffer preferences and to query the current buffer state
+- `DownloadFinishedEvent` to indicate once a download was finished
```
Comment (rolandkakonyi) [inline CHANGELOG.md:9]: nit: What do you think of rewording this a little bit? It flows better for me this way.

```suggestion
- `DownloadFinishedEvent` to signal when the download of specific content has finished
```

File: src/components/PlayerView/events.ts [hunk 2]
```diff
@@ -346,6 +347,10 @@ interface EventProps {
    * Emitted when the current video playback quality has changed.
    */
   onVideoPlaybackQualityChanged: VideoPlaybackQualityChangedEvent;
+  /**
+   * Emitted when a download was finished.
+   */
+  onDownloadFinished: DownloadFinishedEvent;
```
Comment (rolandkakonyi) [inline src/components/PlayerView/events.ts:353]: Please reorder `onDownloadFinished` to match the current alphabetically ordered event listeners here.

**Review by rolandkakonyi — COMMENTED** [review]
Manual testing looks good! 💪

Please see my comment about the code changes.

File: src/components/PlayerView/events.ts (hunk 2 above)
Comment (123mpozzi) [reply src/components/PlayerView/events.ts]: Address in 88ef0dfe897b6d62ebd12d5e917bfd4bb454ac2d

**Review by rolandkakonyi — APPROVED** [review]

**Review by rolandkakonyi — APPROVED** [review]

**Review by zigavehovec — APPROVED** [review]
Nice job 👌 I also tested on Android and everything looks good 🙂

**Pull request merged**